README.md

```diff
@@ -1,2 +1,229 @@
+        ELSE 'Unknown'
+    END AS "CVSS Severity",
+    -- Each vulnerability ID is a different record, we want to count those.
+    COUNT(DISTINCT id) as "Vulnerability Count",
+    -- This average is useful to see the general trend for the data.
+    AVG("Vulnerability Count") OVER (
+        PARTITION BY "CVSS Severity" ORDER BY "Date" ROWS BETWEEN 2 PRECEDING AND CURRENT ROW
+    ) AS "3 day average"
+FROM
+    vulnerability
+WHERE
+    -- Only show data for the past 90 days.
+    DATE_TRUNC('DAY', ingestion_date) > CURRENT_DATE - INTERVAL '90 days'
+GROUP BY "Date", "CVSS Severity"
+ORDER BY "Date"
+```
+
+### Top widespread vulnerable applications
+
+With this view, it’s easy to visualize if a set of applications, in particular, introduces the most risk to the organization. It can help prioritize efforts into which applications to manage.
+
+```sql
+SELECT TOP 10
+	-- We're not using vendor in the graph, but would be just as easy to see vendor data if we were interested in it.
+	applicationvendor,
+	applicationname,
+	-- This will give us the number of endpoints where the vulnerable application is deployed.
+	COUNT(DISTINCT endpoint_data_id) as "Vulnerable Endpoints"
+FROM
+	vulnerability
+WHERE
+	-- For this query, we're only interested in the data from the latest report.
+	ingestion_date = (SELECT MAX(ingestion_date) FROM vulnerability)
+GROUP BY
+	applicationvendor,
+	applicationname
+ORDER BY
+	"Vulnerable Endpoints"
+DESC
+```
+
+### Vulnerability age through time
+
+This view provides interesting data about the vulnerability age in the environment. The graph tends to climb steadily because each day vulnerabilities become a day older, but it can give you some insights into how old the unaddressed vulnerabilities are.
+
+```sql
+SELECT
+	-- Calculate percentiles using days since the vulnerability was detected.
+	PERCENTILE_DISC(0.5) WITHIN GROUP(ORDER BY daysdetected) AS p50,
+	PERCENTILE_DISC(0.95) WITHIN GROUP(ORDER BY daysdetected) AS p95,
+	MAX(daysdetected::INTEGER) AS "Max",
+	DATE_TRUNC('DAY', ingestion_date) AS "Date"
+FROM
+	vulnerability
+WHERE
+	-- Only show data for the past 90 days.
+	DATE_TRUNC('DAY', ingestion_date) > DATE_TRUNC('DAY', CURRENT_DATE - INTERVAL '90 days')
+GROUP BY
+	-- Group by date so we can see how the numbers change daily and look at trends over time.
+	"Date"
+ORDER BY
+	"Date"
+```
+
+### Widespread managed vulnerable applications out of SLA
+
+The view produced by the following query is what we've found most impactful at Canva. It’s more complex than previous views, using an external table to determine which applications we centrally manage. Since non-managed applications are excluded from results, it should, in theory, return no data. Realistically, however, we should expect to see spikes from vulnerabilities being published and then quickly resolved throughout the fleet automatically. Any sustained numbers indicate issues with the vulnerability management process.
```

Comment:
```suggestion
The view produced by the following query is what we've found most impactful at Canva. It’s more complex than previous views, using an external table to determine which applications we centrally manage. Because non-managed applications are excluded from results, it should, in theory, return no data. Realistically, however, we expect spikes from vulnerabilities being published and then resolved automatically throughout the fleet. Any sustained numbers indicate issues with the vulnerability management process.
```